<?xml version="1.0" encoding="UTF-8"?>
<launch>
        <!--  ************** Global Parameters ***************  -->
        <!--param name="/use_sim_time" value="true"/-->

        <!--  ************** Stage Simulator ***************  -->
	<!--node pkg="stage_ros" type="stageros" name="stageros" args="$(find teb_local_planner_tutorials)/stage/maze_carlike.world">
		<remap from="base_scan" to="scan"/>
  	</node-->

        <!--  ************** Navigation ***************  -->
	<node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
    	<rosparam file="$(find cavecrawler_planning)/params/local_costmap_params.yaml" command="load" />
    	<rosparam file="$(find cavecrawler_planning)/params/global_costmap_params.yaml" command="load" />
    	<rosparam file="$(find cavecrawler_planning)/params/costmap_common_params.yaml" command="load" ns="global_costmap" />
    	<rosparam file="$(find cavecrawler_planning)/params/costmap_common_params.yaml" command="load" ns="local_costmap" />
  		<rosparam file="$(find cavecrawler_planning)/params/teb_local_planner_params.yaml" command="load" />

		<!--<param name="base_global_planner" value="global_planner/GlobalPlanner" />
		<param name="planner_frequency" value="1.0" />
		<param name="planner_patience" value="5.0" /> -->

		<param name="base_global_planner" value="ompl_global_planner/OmplGlobalPlanner"/>
		<!-- The params to be tuned for the rrt star global planner -->
		<param name="max_footprint_cost" value="256.0" />
		<param name="relative_validity_check_resolution_" value="0.05" />
        <param name="goal_threshold" value="0.1"/>
        <param name="solver_time" value="1.5"/>

		<param name="base_local_planner" value="teb_local_planner/TebLocalPlannerROS" />
		<param name="controller_frequency" value="5.0" />
		<param name="controller_patience" value="15.0" />
        <param name="clearing_rotation_allowed" value="true"/>
	</node>


    <!--  ****** Maps *****  -->
	<!--node name="map_server" pkg="map_server" type="map_server" args="$(find teb_local_planner_tutorials)/maps/maze.yaml" output="screen">
		<param name="frame_id" value="/map"/>
	</node-->

	<node pkg="amcl" type="amcl" name="amcl" output="screen">
		<rosparam file="$(find cavecrawler_planning)/params/amcl_params.yaml" command="load" />

        <param name="initial_pose_x"            value="0"/>
		<param name="initial_pose_y"            value="0"/>
		<param name="initial_pose_a"            value="0"/>
	</node>

	  <!--  **************** Visualisation ****************  -->
	  <!--node name="rviz" pkg="rviz" type="rviz" args="-d $(find teb_local_planner_tutorials)/cfg/rviz_navigation.rviz"/-->
</launch>